medallion/backends/mongodb_backend.py

```diff
@@ -27,16 +27,21 @@ def _update_manifest(self, new_obj, api_root, collection_id):
         manifest_info = api_root_db["manifests"]
         entry = manifest_info.find_one({"collection_id": collection_id, "id": new_obj["id"]})
         if entry:
-            entry["versions"].append(new_obj["modified"])
-            manifest_info.update_one({"collection_id": collection_id, "id": new_obj["id"]}, {"$set": {"versions": entry["versions"]}})
-            return
-        manifest_info.insert_one({"id": new_obj["id"],
-                                  "collection_id": collection_id,
-                                  "date_added": format_datetime(get_timestamp()),
-                                  "versions": [new_obj["modified"]],
-                                  # hardcoded for now
-                                  "media_types": ["application/vnd.oasis.stix+json; version=2.0"]
-                                  })
+            if "modified" in new_obj:
+                entry["versions"].append(new_obj["modified"])
+                manifest_info.update_one({"collection_id": collection_id, "id": new_obj["id"]}, {"$set": {"versions": entry["versions"]}})
+            # if the new_obj is there, and it has no modified property, then it is immutable, and there is nothing to do.
+        else:
+            if "modified" in new_obj:
```

Comment:
suggest changing these four lines to `version = new_obj.get('modified', new_obj['created'])` and/or adding a comment.

Reply:
Or, pull this out into a small utility function, which can also be called from up above where this same pattern occurs.